federation/schema_syncer.go

```diff
@@ -0,0 +1,73 @@
+package federation
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/samsarahq/go/oops"
+	"github.com/samsarahq/thunder/graphql/introspection"
+)
+
+// SchemaSyncer has a function that checks if the schema has changed,
+// and if so updates the planner in the federated executor
+type SchemaSyncer interface {
+	FetchPlanner(ctx context.Context, optionalArgs interface{}) (*Planner, error)
+	GetRefetchInterval(ctx context.Context) (int64, error)
+}
+type IntrospectionSchemaSyncer struct {
+	executors    map[string]ExecutorClient
+	optionalArgs interface{}
+}
+
+func newIntrospectionSchemaSyncer(ctx context.Context, executors map[string]ExecutorClient, optionalArgs interface{}) *IntrospectionSchemaSyncer {
```

Comment:
Do you think we want to make it public?

Reply:
And maybe add a comment to describe the syncer?

Reply:
We can, although I suspect it will never be used since its always set as the default syncer and a developer can choose to write their own syncer if needed